### ISSUE TITLE
KAlite installation on RaspberryPI

### ISSUE THREAD

**Issue by JMNB** [learningequality/ka-lite#4789]
I want to install KA Lite on a RaspberryPI and followed the instructions in the Installation Guide for version 0.15, using the recommended user id 'pi' as the admin account. When I copied the previous donwloaded video's to the content-directory, they did not show up in Kalite, as is also stated as a bug (issue  #4781).
In one of the comments of issue 4781 as work-around is given to use version 0.13 instead.

I found the instructions to download and install KAlite version 0.13 at https://ka-lite.readthedocs.org/en/0.13.x/installguide/install_linux.html
It went nice until point 8: 'Run the install script with ./setup_unix.sh.'
The setup_unix.sh could not be found.
I search on internet for other instructions, but nowhere I could find a proper workaround.

Thank you beforehand for any help me to find a solution, so I am able to install KAlite on a Raspberry, 

Julia

**Comment by benjaoming** [comment]
Hi @JMNB thanks for writing.

We would like to help you as much as we can, but we simply don't have resources to fix problems with 0.13 -- would you like to help us get the original issue #4781 solved instead?

I would be happy to help, given best-possible guidance to recreate your issue. Could you provide the steps you did over in #4781 ?

I'm wondering:

1) Are you using the ka-lite-raspberry-pi debian installation?
2) How did you originally download the videos that you are moving into the content folder?
3) Have you examined read permissions on the video files?
4) Have you tried running (as the pi user) `kalite videoscan --commandline` ?

**Comment by JMNB** [comment]
Dear Benjaoming, 

 

We are willing to help you! We want to use the RaspberryPI’s with KAlite on a Highschool with 400 students in Addis Ababa in Ethiopia. Especially for Physics, Biology, Chemistry and Math. We like to use this set-up because hardware is low cost, and KALite does not need to have an internet connection.

We have also KAlite 0.15 running on a Windows PC, where we downloaded the videos per topic.

 

Please find below some details: 

*         Hardware: RaspberryPI Model 2B

*         Installation of Raspbian a community created port of Debian Jessie

*         After installation of Raspbian:

o   sudo apt-get update

o   sudo apt-get upgrade

o   reboot

*         Installation of KALite as described on http://ka-lite.readthedocs.org/en/0.15.x/installguide/install_all.html#raspberry-pi

*         Admin userid: pi; KAlite is started automatically at start-up and the assessments are downloaded

*         When KAlite started for the first time, using the pi-admin account, the device registered as ‘one click registration’ and a facility added.

*         Compared to the KAlite on windows, the ‘Default facility’ cannot be deleted on the RaspberryPI

*         We have to select first a facility in the MANAGE tab, before we can download any video.

*         After selecting a video for download on the RaspberryPi, KALite displayes:

o   ‘Launched download  process successfully’

o   The two bars for the progress with ‘LOADING’ and ’Overall progress….’, but no video is downloaded

*         Copying some video lessons, downloaded on the windows pc, to the content-directory and rebooting the RaspberryPI, the videos are not visible in KAlite.

*        Finally I tried to enter the command ‘kalite videoscan –commandline’, but this was not successful. I found out that the should be something like: sudo su pi –s kalite manage videoscan??

 

If you can give me the correct command I will do another try.

 

In September last year I was able to install KAlite successfully on the Raspberry . . .

 

Thanks for your help and I look forward to your answer, 

 

Julia Nolles

 

 

Van: benjaoming [mailto:notifications@github.com] 
Verzonden: 27 January 2016 17:30
Aan: learningequality/ka-lite <ka-lite@noreply.github.com>
CC: JMNB <julia.nolles@me.com>
Onderwerp: Re: [ka-lite] KAlite installation on RaspberryPI (#4789)

 

Hi @JMNB <https://github.com/JMNB>  thanks for writing.

We would like to help you as much as we can, but we simply don't have resources to fix problems with 0.13 -- would you like to help us get the original issue #4781 <https://github.com/learningequality/ka-lite/issues/4781>  solved instead?

I would be happy to help, given best-possible guidance to recreate your issue. Could you provide the steps you did over in #4781 <https://github.com/learningequality/ka-lite/issues/4781>  ?

I'm wondering:

1) Are you using the ka-lite-raspberry-pi debian installation?
2) How did you originally download the videos that you are moving into the content folder?
3) Have you examined read permissions on the video files?
4) Have you tried running (as the pi user) kalite videoscan --commandline ?

**Comment by JMNB** [comment]
Dear Benjaoming, 

 

Today I did another try installing KAlite 0.15 on the RaspberryPi. 

After installation and copying a number of videos in the content folder, when starting up KA lite, still no videos showed up. 

 

But in the download tree, those videos I copied were indicated as downloaded . . .

 

Downloading other videos was still not possible. The same message ‘Launched download  process successfully’ was displayed. But in one way or the other a new MP4 file was created in the content directory (size remains 0 bytes)

 

Next I tried to delete one of the videos in KAlite, but after click on yes on the confirmation question, nothing happened.

 

Please let me know if you have a new version, I am willing to test it for you. If you show me how to download and install the new version. We really would like KALite 0.15 running on a Raspberry for the Selam High School in Ethiopia (around 400 students).

 

Thanks for your help and support beforehand, 

 

Wim and Julia Nolles

 

 

Van: Julia Nolles [mailto:julia.nolles@me.com] 
Verzonden: 28 January 2016 12:06
Aan: 'learningequality/ka-lite' <reply@reply.github.com>
Onderwerp: RE: [ka-lite] KAlite installation on RaspberryPI (#4789)

 

Dear Benjaoming, 

 

We are willing to help you! We want to use the RaspberryPI’s with KAlite on a Highschool with 400 students in Addis Ababa in Ethiopia. Especially for Physics, Biology, Chemistry and Math. We like to use this set-up because hardware is low cost, and KALite does not need to have an internet connection.

We have also KAlite 0.15 running on a Windows PC, where we downloaded the videos per topic.

 

Please find below some details: 

*         Hardware: RaspberryPI Model 2B

*         Installation of Raspbian a community created port of Debian Jessie

*         After installation of Raspbian:

o   sudo apt-get update

o   sudo apt-get upgrade

o   reboot

*         Installation of KALite as described on http://ka-lite.readthedocs.org/en/0.15.x/installguide/install_all.html#raspberry-pi

*         Admin userid: pi; KAlite is started automatically at start-up and the assessments are downloaded

*         When KAlite started for the first time, using the pi-admin account, the device registered as ‘one click registration’ and a facility added.

*         Compared to the KAlite on windows, the ‘Default facility’ cannot be deleted on the RaspberryPI

*         We have to select first a facility in the MANAGE tab, before we can download any video.

*         After selecting a video for download on the RaspberryPi, KALite displayes:

o   ‘Launched download  process successfully’

o   The two bars for the progress with ‘LOADING’ and ’Overall progress….’, but no video is downloaded

*         Copying some video lessons, downloaded on the windows pc, to the content-directory and rebooting the RaspberryPI, the videos are not visible in KAlite.

*        Finally I tried to enter the command ‘kalite videoscan –commandline’, but this was not successful. I found out that the should be something like: sudo su pi –s kalite manage videoscan??

 

If you can give me the correct command I will do another try.

 

In September last year I was able to install KAlite successfully on the Raspberry . . .

 

Thanks for your help and I look forward to your answer, 

 

Julia Nolles

 

 

Van: benjaoming [ <mailto:notifications@github.com> mailto:notifications@github.com] 
Verzonden: 27 January 2016 17:30
Aan: learningequality/ka-lite < <mailto:ka-lite@noreply.github.com> ka-lite@noreply.github.com>
CC: JMNB < <mailto:julia.nolles@me.com> julia.nolles@me.com>
Onderwerp: Re: [ka-lite] KAlite installation on RaspberryPI (#4789)

 

Hi @JMNB <https://github.com/JMNB>  thanks for writing.

We would like to help you as much as we can, but we simply don't have resources to fix problems with 0.13 -- would you like to help us get the original issue #4781 <https://github.com/learningequality/ka-lite/issues/4781>  solved instead?

I would be happy to help, given best-possible guidance to recreate your issue. Could you provide the steps you did over in #4781 <https://github.com/learningequality/ka-lite/issues/4781>  ?

I'm wondering:

1) Are you using the ka-lite-raspberry-pi debian installation?
2) How did you originally download the videos that you are moving into the content folder?
3) Have you examined read permissions on the video files?
4) Have you tried running (as the pi user) kalite videoscan --commandline ?

**Comment by JMNB** [comment]
Dear Benjaoming, 

 

After a while (I don’t know how long), an error message was displayed on the screen, after trying to delete a video file:

Unexpected error, contact FLE with the following information: 504

There was an unexpected error

[object OBJECT]

 

Perhaps it is valuable information for you.

 

Julia

 

Van: Julia Nolles [mailto:julia.nolles@me.com] 
Verzonden: 05 February 2016 21:38
Aan: 'learningequality/ka-lite' <reply@reply.github.com>
Onderwerp: RE: [ka-lite] KAlite installation on RaspberryPI (#4789)

 

Dear Benjaoming, 

 

Today I did another try installing KAlite 0.15 on the RaspberryPi. 

After installation and copying a number of videos in the content folder, when starting up KA lite, still no videos showed up. 

 

But in the download tree, those videos I copied were indicated as downloaded . . .

 

Downloading other videos was still not possible. The same message ‘Launched download  process successfully’ was displayed. But in one way or the other a new MP4 file was created in the content directory (size remains 0 bytes)

 

Next I tried to delete one of the videos in KAlite, but after click on yes on the confirmation question, nothing happened.

 

Please let me know if you have a new version, I am willing to test it for you. If you show me how to download and install the new version. We really would like KALite 0.15 running on a Raspberry for the Selam High School in Ethiopia (around 400 students).

 

Thanks for your help and support beforehand, 

 

Wim and Julia Nolles

 

 

 

Van: Julia Nolles [mailto:julia.nolles@me.com] 
Verzonden: 28 January 2016 12:06
Aan: 'learningequality/ka-lite' <reply@reply.github.com>
Onderwerp: RE: [ka-lite] KAlite installation on RaspberryPI (#4789)

 

Dear Benjaoming, 

 

We are willing to help you! We want to use the RaspberryPI’s with KAlite on a Highschool with 400 students in Addis Ababa in Ethiopia. Especially for Physics, Biology, Chemistry and Math. We like to use this set-up because hardware is low cost, and KALite does not need to have an internet connection.

We have also KAlite 0.15 running on a Windows PC, where we downloaded the videos per topic.

 

Please find below some details: 

*         Hardware: RaspberryPI Model 2B

*         Installation of Raspbian a community created port of Debian Jessie

*         After installation of Raspbian:

o   sudo apt-get update

o   sudo apt-get upgrade

o   reboot

*         Installation of KALite as described on http://ka-lite.readthedocs.org/en/0.15.x/installguide/install_all.html#raspberry-pi

*         Admin userid: pi; KAlite is started automatically at start-up and the assessments are downloaded

*         When KAlite started for the first time, using the pi-admin account, the device registered as ‘one click registration’ and a facility added.

*         Compared to the KAlite on windows, the ‘Default facility’ cannot be deleted on the RaspberryPI

*         We have to select first a facility in the MANAGE tab, before we can download any video.

*         After selecting a video for download on the RaspberryPi, KALite displayes:

o   ‘Launched download  process successfully’

o   The two bars for the progress with ‘LOADING’ and ’Overall progress….’, but no video is downloaded

*         Copying some video lessons, downloaded on the windows pc, to the content-directory and rebooting the RaspberryPI, the videos are not visible in KAlite.

*        Finally I tried to enter the command ‘kalite videoscan –commandline’, but this was not successful. I found out that the should be something like: sudo su pi –s kalite manage videoscan??

 

If you can give me the correct command I will do another try.

 

In September last year I was able to install KAlite successfully on the Raspberry . . .

 

Thanks for your help and I look forward to your answer, 

 

Julia Nolles

 

 

Van: benjaoming [ <mailto:notifications@github.com> mailto:notifications@github.com] 
Verzonden: 27 January 2016 17:30
Aan: learningequality/ka-lite < <mailto:ka-lite@noreply.github.com> ka-lite@noreply.github.com>
CC: JMNB < <mailto:julia.nolles@me.com> julia.nolles@me.com>
Onderwerp: Re: [ka-lite] KAlite installation on RaspberryPI (#4789)

 

Hi @JMNB <https://github.com/JMNB>  thanks for writing.

We would like to help you as much as we can, but we simply don't have resources to fix problems with 0.13 -- would you like to help us get the original issue #4781 <https://github.com/learningequality/ka-lite/issues/4781>  solved instead?

I would be happy to help, given best-possible guidance to recreate your issue. Could you provide the steps you did over in #4781 <https://github.com/learningequality/ka-lite/issues/4781>  ?

I'm wondering:

1) Are you using the ka-lite-raspberry-pi debian installation?
2) How did you originally download the videos that you are moving into the content folder?
3) Have you examined read permissions on the video files?
4) Have you tried running (as the pi user) kalite videoscan --commandline ?

**Comment by benjaoming** [comment]
Hi @JMNB 

Thanks so much for your email and thorough feedback. Sorry that I haven't responded until now.

You've caught me in a bit of a limbo regarding the RPi issues because we have 0.16 on the doorstep. For sure, it seems that 0.15.1 has a blocking video download and directory scanning issue that warrants immediate attention. I'll discuss this with the team, and hopefully I can attend to this Monday and Tuesday and get an RPi fix release ready.

**1. The download job scheduler issue**

>  Downloading other videos was still not possible. The same message ‘Launched download process successfully’ was displayed. But in one way or the other a new MP4 file was created in the content directory (size remains 0 bytes)

This is where I'm at, too. I've installed KA Lite 0.15.1 on Raspbian Jessy and I get exactly the same result. I managed to download a single video file. I think that old issues are haunting us and quite likely it's due to the somewhat "DIY" job scheduler that downloads all the videos.

**2. Scanning for downloaded videos**

>  Finally I tried to enter the command ‘kalite videoscan –commandline’, but this was not successful. I found out that the should be something like: sudo su pi –s kalite manage videoscan??

That's true, since the process runs with the "pi" user, you have to be running `kalite` commands with this user. Did this work for you or is the issues still extant?

**3. Deleting videos**

>  Next I tried to delete one of the videos in KAlite, but after click on yes on the confirmation question, nothing happened. 

That one is strange, I cannot recreate it. If the server was stopped or killed or otherwise busy, that could explain it.. otherwise this needs further investigation.

**Comment by JMNB** [comment]
Dear Benjamin, 

 

Thank you for your answer. 

 

Today I discovered that the video I wanted to delete, was deleted in the end, as it showed up in the tree on the download video page.

 

I Indicated that this file must be downloaded. During the dowload (unsuccessful in the end) I just checked if it was possible to watch the videos and to my surprise I was able to start a video. . . 

When I did another, whole Kalite was not responding anymore and I shutdown the Raspberry. When I reboot the Raspberry again, Kalite refused to start –up again. 

 

Now I am reinstalling the Raspberry, and will install also Kalite again. 

 

If I can be of any help testing KALite on a Raspberry, please let me know.

 

Success and  kind regards, 

 

Julia

 

Van: Benjamin Bach [mailto:notifications@github.com] 
Verzonden: 10 February 2016 20:08
Aan: learningequality/ka-lite <ka-lite@noreply.github.com>
CC: JMNB <julia.nolles@me.com>
Onderwerp: Re: [ka-lite] KAlite installation on RaspberryPI (#4789)

 

Hi @JMNB <https://github.com/JMNB>  

Thanks so much for your email and thorough feedback. Sorry that I haven't responded until now.

You've caught me in a bit of a limbo regarding the RPi issues because we have 0.16 on the doorstep. For sure, it seems that 0.15.1 has a blocking video download and directory scanning issue that warrants immediate attention. I'll discuss this with the team, and hopefully I can attend to this Monday and Tuesday and get an RPi fix release ready.

1. The download job scheduler issue

Downloading other videos was still not possible. The same message ‘Launched download process successfully’ was displayed. But in one way or the other a new MP4 file was created in the content directory (size remains 0 bytes)

This is where I'm at, too. I've installed KA Lite 0.15.1 on Raspbian Jessy and I get exactly the same result. I managed to download a single video file. I think that old issues are haunting us and quite likely it's due to the somewhat "DIY" job scheduler that downloads all the videos.

2. Scanning for downloaded videos

Finally I tried to enter the command ‘kalite videoscan –commandline’, but this was not successful. I found out that the should be something like: sudo su pi –s kalite manage videoscan??

That's true, since the process runs with the "pi" user, you have to be running kalite commands with this user. Did this work for you or is the issues still extant?

3. Deleting videos

Next I tried to delete one of the videos in KAlite, but after click on yes on the confirmation question, nothing happened. 

That one is strange, I cannot recreate it. If the server was stopped or killed or otherwise busy, that could explain it.. otherwise this needs further investigation.

**Comment by benjaoming** [comment]
Thanks so much, Julia! I will stay in touch, please let me know if you
encounter any other oddities, your explanations are very thorough and easy
to work with :+1:

**Comment by JMNB** [comment]
Dear Benjamin, 

 

Now I followed another process after downloading the exercises:

1.       Copy of some video files to the content directory

2.       Command : KAlite manage setup

3.       Shut down and reboot RaspberryPI

4.       Start KAlite browser (127.0.0.1:8008)

5.       Create administrator account pi

6.       Login with account pi

7.       Renamed default facility

8.       Check if the copied video lectures showed up in the download tree on the MANAGE tab

9.       Start the LEARN tab

10.   To my surprise it showed me that all video lectures were available in the KAlite; no title was faded out

11.   But when starting one video the message ‘Unexpected error; contact FLE with the following information: 504 . . . ‘ was displayed. If the video was available in the content directory or not.

 

And by the way, the exercises are doing fine!

 

Julia

 

Van: Benjamin Bach [mailto:notifications@github.com] 
Verzonden: 10 February 2016 21:18
Aan: learningequality/ka-lite <ka-lite@noreply.github.com>
CC: JMNB <julia.nolles@me.com>
Onderwerp: Re: [ka-lite] KAlite installation on RaspberryPI (#4789)

 

Thanks so much, Julia! I will stay in touch, please let me know if you
encounter any other oddities, your explanations are very thorough and easy
to work with :+1:

**Comment by benjaoming** [comment]
Hi Julia,

The exercises should be fine as they are bundled with the installation.

Maybe you can have a look in /home/pi/.kalite/server.log to see if there
are any interesting error messages?

Best,
Ben

**Comment by JMNB** [comment]
Dear Benjamin, good morning.

 

Please find attached the server.log file, as I copied it from the Raspberry. 

 

I don’t know if there any interesting error messages.

 

Kind regards, 

 

Julia

 

 

Van: Benjamin Bach [mailto:notifications@github.com] 
Verzonden: 10 February 2016 23:03
Aan: learningequality/ka-lite <ka-lite@noreply.github.com>
CC: JMNB <julia.nolles@me.com>
Onderwerp: Re: [ka-lite] KAlite installation on RaspberryPI (#4789)

 

Hi Julia,

The exercises should be fine as they are bundled with the installation.

Maybe you can have a look in /home/pi/.kalite/server.log to see if there
are any interesting error messages?

Best,
Ben

**Comment by benjaoming** [comment]
Hi Julia,

This conversation is actually facilitated through a Github issue, meaning
that attachments are removed. But maybe you can place it here:
https://pastee.org/

**Comment by JMNB** [comment]
No problem, please find the content of the server-log file below, Julia

 

[11/Feb/2016:13:37:49] ENGINE Listening for SIGHUP.

INFO:cherrypy.error:

[11/Feb/2016:13:37:49] ENGINE Listening for SIGHUP.

 

[11/Feb/2016:13:37:49] ENGINE Listening for SIGTERM.

INFO:cherrypy.error:

[11/Feb/2016:13:37:49] ENGINE Listening for SIGTERM.

 

[11/Feb/2016:13:37:49] ENGINE Listening for SIGUSR1.

INFO:cherrypy.error:

[11/Feb/2016:13:37:49] ENGINE Listening for SIGUSR1.

 

[11/Feb/2016:13:37:49] ENGINE Bus STARTING

INFO:cherrypy.error:

[11/Feb/2016:13:37:49] ENGINE Bus STARTING

 

[11/Feb/2016:13:37:49]  Loading and serving the Django application

INFO:cherrypy.error:

[11/Feb/2016:13:37:49]  Loading and serving the Django application

 

[11/Feb/2016:13:37:49] ENGINE Started monitor thread '_TimeoutMonitor'.

INFO:cherrypy.error:

[11/Feb/2016:13:37:49] ENGINE Started monitor thread '_TimeoutMonitor'.

INFO:kalite:Running 1 due jobs... ("Secure Sync")

 

[11/Feb/2016:13:37:49] ENGINE Serving on 0.0.0.0:7007

INFO:cherrypy.error:

[11/Feb/2016:13:37:49] ENGINE Serving on 0.0.0.0:7007

 

[11/Feb/2016:13:37:49] ENGINE Bus STARTED

INFO:cherrypy.error:

[11/Feb/2016:13:37:49] ENGINE Bus STARTED

WARNING:py.warnings:/usr/share/kalite/python-packages/fle_utils/set_process_priority.py:78: DeprecationWarning: set_ionice() is deprecated; use ionice() instead

  this_process.set_ionice(psutil.IOPRIO_CLASS_IDLE)

 

 

Van: Benjamin Bach [mailto:notifications@github.com] 
Verzonden: 11 February 2016 17:11
Aan: learningequality/ka-lite <ka-lite@noreply.github.com>
CC: JMNB <julia.nolles@me.com>
Onderwerp: Re: [ka-lite] KAlite installation on RaspberryPI (#4789)

 

Hi Julia,

This conversation is actually facilitated through a Github issue, meaning
that attachments are removed. But maybe you can place it here:
https://pastee.org/

**Comment by benjaoming** [comment]
Could you verify that it got sent correctly, something is odd about "INFO:cherrypy.error:" and then nothing follows.. normally our error logging contains some kind of useful information :)

**Comment by JMNB** [comment]
Benjamin, 

 

But I did something wrong, but I restarted the Raspberry twice and the same content in the server.log file was written.

 

Do I need to also start the web application?

Or any other command?

 

JUlia

 

Van: Benjamin Bach [mailto:notifications@github.com] 
Verzonden: 11 February 2016 18:47
Aan: learningequality/ka-lite <ka-lite@noreply.github.com>
CC: JMNB <julia.nolles@me.com>
Onderwerp: Re: [ka-lite] KAlite installation on RaspberryPI (#4789)

 

Could you verify that it got sent correctly, something is odd about "INFO:cherrypy.error:" and then nothing follows.. normally our error logging contains some kind of useful information :)

**Comment by JMNB** [comment]
Benjamin, I just tried something – perhaps helpful.

I stop the KAlite server (kalite stop) and the message kalite stopped was displayed.

 

After entering kalite start, the following was displayed on the screen:

 

Running ‘kalite start ‘ as deamon (system service)

 

Stand while the server loads its data. . . 

 

Port 8008 is occupied. Please close the process that is using it

 

And no new messages are written into the server.log file?!

 

I do not know how to stop the process which occupying port 8008.

 

Julia

 

Van: Benjamin Bach [mailto:notifications@github.com] 
Verzonden: 11 February 2016 18:47
Aan: learningequality/ka-lite <ka-lite@noreply.github.com>
CC: JMNB <julia.nolles@me.com>
Onderwerp: Re: [ka-lite] KAlite installation on RaspberryPI (#4789)

 

Could you verify that it got sent correctly, something is odd about "INFO:cherrypy.error:" and then nothing follows.. normally our error logging contains some kind of useful information :)

**Comment by benjaoming** [comment]
This is because the way that kalite is started is through "sudo service
ka-lite start" which will automatically `add --port=7007` such that the
actual start command becomes `kalite start --port=7007`. For different
reasons, we've moved kalite to port 7007, and then nginx is running on 8008
and intercepts any request for static media such as videos. It's much fast
using nginx, which is critical on Raspberry Pi.

**Comment by synetics** [comment]
Hi Benjaoming
I have a fresh install of kalite 0.15.1 on Raspberry Pi 3, I can view the exercises, but no video playback. I was able to download new Videos, I could sync with a restart. But still no video will play. On another tab, I can directly play the MP4 video as a file. 
So the problem must be in the way the Javascript is calling the MP4 files. 
Any clue as to 504 error. I am also getting that.
What can we check and give you some clue for fixing this.
I am also supporting a school with Khan Academy.
Thanks.
Joseph Mathew

**Comment by synetics** [comment]
Hi Benjaoming
Here is a temporary solution, until you find the problem and fix it.
I changed the settings to:
kalite.project.settings.base
This worked and I could see the videos. 
I would request Julia to try it:
File name is: ~/.kalite/settings.py
The old line is: from kalite.project.settings.raspberry_pi import *
The new line is: from kalite.project.settings.base import *
Save it, restart the server, that should should show 
Running videoscan
Preloading exercise data from language en
Preloading content data from language en
Preloading topic tree data for language en

and messages related to IP address and port.
Try it and post your result.
Thanks
Joseph Mathew

**Comment by JMNB** [comment]
Dear Benjamin and Joseph, 

Now almost two months later, we have returned from Addis where we installed KAlite 0:14 on a high school. For the time being no RapsberryPI is left at home, where we can do the tests.

Sorry, 

Kind regards, 
Julia

**Comment by benjaoming** [comment]
Hi @JMNB and @synetics 

Looking forwards, we are going to do a bug fixed release of 0.16.6 and another 0.17 release with a minimal set of new features.

I would like to honor the efforts in this issue somehow, but it's a lot of stuff piled together. Trying to boil it down:

 * Some issues with `DeprecationWarning: set_ionice() is deprecated; use ionice() instead` - this should be okay as in the future we have decoupled cherrypy form the source distribution so we can upgrade it from upstream.

 * We need a proper guide to installing Raspberry Pis.

 * There has been seemingly both issues with broken download interfaces and senseless error message feedback in 0.15. This has affected RPI installations very negatively, and I think it's safe to conclude that 0.15 is unfit for RPI, which is also the conclusion in other experiences.

It's a little hard to reuse the information about broken-ness in 0.15, as most of these components have now gratefully been replaced in 0.16.

The current version of 0.16.5 has some known issues regarding translations and a few exercises (not many remaining!) are suffering from issues caused by Khan Academy changing data APIs and updating JS exercises framework. We're working on that for a 0.16.6 patch release.

But for now, I can say that I have a very smooth installation of 0.16.5 running on a RPI next to me, and since we've had it out for about a month during which I haven't heard of any RPI-related problems at this point, although I'm sure they'll show up :)

I'm keeping the issue open for now as an example of why we need proper documentation, I hope we will be able to close it with a good HOWTO section for the docs.

**Comment by benjaoming** [comment]
The background of why I might have sounded like someone lowering expectations for future KA Lite releases is just to be understood as a decision to focus all efforts on a much better platform :) So it's all in a good nature. It's called [Kolibri](https://learningequality.org/kolibri/) and a first version will be out in late 2016.

**Comment by benjaoming** [comment]
Made some improvement regarding adding the PPA, but we still need an overall guide for setting up KA Lite on a Raspberry Pi.

Closing this in favour of a simpler issue.

Thanks all for your feedback!
